qhub/deploy.py

```diff
@@ -127,3 +131,32 @@ def add_clearml_dns(zone_name, record_name, record_type, ip_or_hostname):
 
     for dns_record in dns_records:
         update_record(zone_name, dns_record, record_type, ip_or_hostname)
+
+
+def check_secrets(config):
+    """
+    Checks that the appropriate variables are set based on the current config.
+    These variables are prefixed with TF_VAR_ and are used to populate the
+    corresponding variables in the terraform deployment. e.g.
+    TF_VAR_prefect_token sets the prefect_token variable in Terraform. These
+    values are set in the terraform state but are not leaked when the
+    terraform render occurs.
+    """
+
+    missing_env_vars = []
+
+    # Check prefect integration set up.
+    if "prefect" in config:
```

Comment:
`prefect` can be in the config, but disabled, like the following

```
prefect:
  enabled: false
```

In this case the `TF_VAR_prefect_token` isn't necessary to be present in the environment.